Lib/test/test_os.py

```diff
@@ -230,6 +230,54 @@ def test_read(self):
             self.assertEqual(type(s), bytes)
             self.assertEqual(s, b"spam")
 
+    def test_readinto(self):
+        with open(os_helper.TESTFN, "w+b") as fobj:
+            fobj.write(b"spam")
+            fobj.flush()
+            fd = fobj.fileno()
+            os.lseek(fd, 0, 0)
+            buffer = bytearray(8)
+            s = os.readinto(fd, buffer)
+            self.assertEqual(type(s), int)
+            self.assertEqual(s, 4)
+            # Should overwrite the first 4 bytes of the buffer.
+            self.assertEqual(bytes(buffer), b"spam\0\0\0\0")
```

Comment:
I don't think that testing for zeros is interesting here:

```suggestion
            self.assertEqual(buffer[:4], b"spam")
```